packages/core/src/controllers/store.ts

```diff
@@ -23,6 +23,10 @@ export class Store<Key, Data extends Record<string, any>> extends IStore<Key, Da
 
   private storagePrefix = CORE_STORAGE_PREFIX;
 
+  // stores recently deleted key to return different rejection message when key is not found
+  private recentlyDeleted: unknown[] = [];
```

Comment:
Seems like the type is known here?

```suggestion
  private recentlyDeleted: Key[] = [];
```